components/kyma-environment-broker/internal/process/provisioning/input/input.go

```diff
@@ -17,8 +17,10 @@ type Config struct {
 }
 
 type RuntimeInput struct {
-	input     gqlschema.ProvisionRuntimeInput
-	overrides map[string][]*gqlschema.ConfigEntryInput
+	input           gqlschema.ProvisionRuntimeInput
+	mutex           *nsync.NamedMutex
```

Comment:
what is the purpose of using named mutex? You are locking access to a field components by different threads. That's ok, but why NamedMutext. Methods SetOverrides and AppendOverrides shares the same field but use different lock name - it works like different mutexes.